### DISCUSSION TITLE
NNX MultiHeadDotProductAttention

### DISCUSSION THREAD

**Discussion by PVirie** [google/flax#4287] [Ideas] [answered]
I've been using flax for a while and it's the best neural network module available as of now, to say the least.

My question is short:
Could you share the road map for an implementation for MultiHeadDotProductAttention module in NNX?
I wish to know when should I decide to upgrade from linen one.

Thank you in advance.

**Comment by PVirie** [comment]
Nvm I saw it in the nnx source.
If you have time, please update the document for how to use. An example would also be nice :)

**Reply by IvyZX** [reply, marked as answer]
Yes we simplified the name to [`MultiHeadAttention`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/nn/attention.html#flax.nnx.MultiHeadAttention). Please check out our [API reference](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/nn/attention.html#flax.nnx.MultiHeadAttention) for how to use it.
